exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/ParquetGroupScan.java

```diff
@@ -268,11 +276,15 @@ private FileSelection expandIfNecessary(FileSelection selection) throws IOExcept
     // use the cacheFileRoot if provided (e.g after partition pruning)
     Path metaFilePath = new Path(cacheFileRoot != null ? cacheFileRoot : selectionRoot, Metadata.METADATA_FILENAME);
     if (!fs.exists(metaFilePath)) { // no metadata cache
+      MetadataContext metaContext = selection.getMetaContext();
```

Comment:
It seems you set metadata context, when there can be the cases when it is still null? Also can you please explain why there are so many checks in the code if meta context is null, can we make sure it always exists?